h2o-py/h2o/explanation/_leaderboard.py

```diff
@@ -0,0 +1,49 @@
+from h2o.automl._base import H2OAutoMLBaseMixin
```

Comment:
Leaderboard is used in `.explain` and I didn't find any better place to put it yet. I am open to moving it anywhere. Where would you suggest to move it?

Reply:
I think we need to try to avoid having top modules that are either too specific or too generic (e.g. `information_retrieval`, `utils`) 
I'm hesitating between top level: `h2o/leaderboard.py` and creating a `scoring` top module: `h2o/scoring/_leaderboard.py`
`scoring` module may be confusing as all the metrics logic is in `h2o/model/*` and I'm struggling finding what else would go there... so top-level makes sense I guess.
wdyt?

Reply:
what about just a  top level`h2o/scoring.py` module? and we put this `make_leaderboard` function there?
I kinda like this